apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/MessagingConfiguration.java

```diff
@@ -95,9 +112,35 @@ public boolean shouldEndMessagingTransactionOnPoll() {
         return endMessagingTransactionOnPoll.get();
     }
 
-    public enum Strategy {
+    public enum JmsStrategy {
```

Comment:
[question] While it's only used for JMS at the moment, could it become relevant to other messaging systems as well ?

Reply:
It's here for VERY long, with a description saying it is specific to JMS. If we need to change it in the future, we can. Note that the config property name was not changed, even though it is not a documented one. This is only an internal `enum`